tests/e2e/testcases/icm/deploy/suite.go

```diff
@@ -0,0 +1,441 @@
+package deploy
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/interchain"
+	"github.com/ava-labs/avalanche-cli/tests/e2e/commands"
+	"github.com/ava-labs/avalanche-cli/tests/e2e/utils"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+const (
+	ewoqKeyName = "ewoq"
+	subnetName  = "testSubnet"
+)
+
+var _ = ginkgo.Describe("[ICM] deploy", func() {
+	ginkgo.Context("with valid input", func() {
+		ginkgo.BeforeEach(func() {
+			commands.StartNetwork()
+		})
+
+		ginkgo.AfterEach(func() {
+			commands.CleanNetwork()
+			err := utils.DeleteConfigs(subnetName)
+			gomega.Expect(err).Should(gomega.BeNil())
+			utils.DeleteCustomBinary(subnetName)
+		})
+		ginkgo.It("should deploy ICM contracts into c-chain", func() {
+			globalFlags := utils.GlobalFlags{
+				"local":             true,
+				"skip-update-check": true,
+			}
+			testFlags := utils.TestFlags{
+				"key": ewoqKeyName,
+			}
+			commandArguments := []string{
+				"--c-chain",
```

Comment:
shouldn't this be a flag

Reply:
anyways seems interesting that using c chain flagand not using it in the test case after this both result in "ICM Messenger successfully deployed to C-Chain".

Should we make it so that it will deploy to C-Chain always even without flag -> this means we remove the flag @felipemadero

Reply:
--c-chain specifies deploy on c-chain only. it is the specification of the l1. it can be either c-chain or some blockchain name so the flag is needed.
but, for local network, c-chain is automatically deployed even if the blockchain name is another

Reply:
this not happens on fuji/mainnet, there you only deploy where indicated